### PR TITLE
Add rules for many packages needed for ROS 2 on FreeBSD

### PR DESCRIPTION
**Pull request by cottsay**
Note that to better handle Python packages which change name when the Python minor version changes, I'm listing the port path instead of the package name. It appears that the package manager handler for FreeBSD in rosdep already handles this correctly.

- asio: https://cgit.freebsd.org/ports/tree/net/asio/Makefile
- benchmark: https://cgit.freebsd.org/ports/tree/devel/benchmark/Makefile
- bullet: https://cgit.freebsd.org/ports/tree/devel/bullet/Makefile
- llvm: https://cgit.freebsd.org/ports/tree/devel/llvm/Makefile
- llvm: https://cgit.freebsd.org/ports/tree/devel/llvm/Makefile
- cppcheck: https://cgit.freebsd.org/ports/tree/devel/cppcheck/Makefile
- file: https://cgit.freebsd.org/ports/tree/sysutils/file/Makefile
- googletest: https://cgit.freebsd.org/ports/tree/devel/googletest/Makefile
- gcc: https://cgit.freebsd.org/ports/tree/lang/gcc/Makefile
- freetype2: https://cgit.freebsd.org/ports/tree/print/freetype2/Makefile
- freetype2: https://cgit.freebsd.org/ports/tree/print/freetype2/Makefile
- lttng-ust: https://cgit.freebsd.org/ports/tree/sysutils/lttng-ust/Makefile
- liblz4: https://cgit.freebsd.org/ports/tree/archivers/liblz4/Makefile
- liblz4: https://cgit.freebsd.org/ports/tree/archivers/liblz4/Makefile
- ncurses: https://cgit.freebsd.org/ports/tree/devel/ncurses/Makefile
- opencv: https://cgit.freebsd.org/ports/tree/graphics/opencv/Makefile
- qt5-svg: https://cgit.freebsd.org/ports/tree/graphics/qt5-svg/Makefile
- sqlite3: https://cgit.freebsd.org/ports/tree/databases/sqlite3/Makefile
- libX11: https://cgit.freebsd.org/ports/tree/x11/libX11/Makefile
- libxml2: https://cgit.freebsd.org/ports/tree/textproc/libxml2/Makefile
- libXrandr: https://cgit.freebsd.org/ports/tree/x11/libXrandr/Makefile
- libyaml: https://cgit.freebsd.org/ports/tree/textproc/libyaml/Makefile
- libyaml: https://cgit.freebsd.org/ports/tree/textproc/libyaml/Makefile
- zstd: https://cgit.freebsd.org/ports/tree/archivers/zstd/Makefile
- openssl: https://cgit.freebsd.org/ports/tree/security/openssl/Makefile
- pybind11: https://cgit.freebsd.org/ports/tree/devel/pybind11/Makefile
- qt5-buildtools: https://cgit.freebsd.org/ports/tree/devel/qt5-buildtools/Makefile
- qt5-qmake: https://cgit.freebsd.org/ports/tree/devel/qt5-qmake/Makefile
- spdlog: https://cgit.freebsd.org/ports/tree/devel/spdlog/Makefile
- uncrustify: https://cgit.freebsd.org/ports/tree/textproc/uncrustify/Makefile
- libyaml: https://cgit.freebsd.org/ports/tree/textproc/libyaml/Makefile
- devel/py-pydocstyle: https://cgit.freebsd.org/ports/tree/devel/py-pydocstyle/Makefile
- devel/py-pyflakes: https://cgit.freebsd.org/ports/tree/devel/py-pyflakes/Makefile
- python3: https://cgit.freebsd.org/ports/tree/lang/python3/Makefile
- devel/py-argcomplete: https://cgit.freebsd.org/ports/tree/devel/py-argcomplete/Makefile
- graphics/py-cairo: https://cgit.freebsd.org/ports/tree/graphics/py-cairo/Makefile
- security/py-cryptography: https://cgit.freebsd.org/ports/tree/security/py-cryptography/Makefile
- textproc/py-empy: https://cgit.freebsd.org/ports/tree/textproc/py-empy/Makefile
- devel/py-fastjsonschema: https://cgit.freebsd.org/ports/tree/devel/py-fastjsonschema/Makefile
- devel/py-flake8: https://cgit.freebsd.org/ports/tree/devel/py-flake8/Makefile
- devel/py-flake8-builtins: https://cgit.freebsd.org/ports/tree/devel/py-flake8-builtins/Makefile
- devel/py-flake8-comprehensions: https://cgit.freebsd.org/ports/tree/devel/py-flake8-comprehensions/Makefile
- devel/py-flake8-docstrings: https://cgit.freebsd.org/ports/tree/devel/py-flake8-docstrings/Makefile
- devel/py-flake8-import-order: https://cgit.freebsd.org/ports/tree/devel/py-flake8-import-order/Makefile
- devel/py-flake8-quotes: https://cgit.freebsd.org/ports/tree/devel/py-flake8-quotes/Makefile
- python3: https://cgit.freebsd.org/ports/tree/lang/python3/Makefile
- python3: https://cgit.freebsd.org/ports/tree/lang/python3/Makefile
- devel/py-lark: https://cgit.freebsd.org/ports/tree/devel/py-lark/Makefile
- devel/py-lxml: https://cgit.freebsd.org/ports/tree/devel/py-lxml/Makefile
- math/py-matplotlib: https://cgit.freebsd.org/ports/tree/math/py-matplotlib/Makefile
- devel/py-mypy: https://cgit.freebsd.org/ports/tree/devel/py-mypy/Makefile
- math/py-numpy: https://cgit.freebsd.org/ports/tree/math/py-numpy/Makefile
- devel/py-packaging: https://cgit.freebsd.org/ports/tree/devel/py-packaging/Makefile
- graphics/py-pillow: https://cgit.freebsd.org/ports/tree/graphics/py-pillow/Makefile
- sysutils/py-psutil: https://cgit.freebsd.org/ports/tree/sysutils/py-psutil/Makefile
- devel/py-pycodestyle: https://cgit.freebsd.org/ports/tree/devel/py-pycodestyle/Makefile
- graphics/py-pydot: https://cgit.freebsd.org/ports/tree/graphics/py-pydot/Makefile
- graphics/py-pygraphviz: https://cgit.freebsd.org/ports/tree/graphics/py-pygraphviz/Makefile
- devel/py-pytest: https://cgit.freebsd.org/ports/tree/devel/py-pytest/Makefile
- devel/py-pytest-cov: https://cgit.freebsd.org/ports/tree/devel/py-pytest-cov/Makefile
- devel/py-pytest-mock: https://cgit.freebsd.org/ports/tree/devel/py-pytest-mock/Makefile
- devel/py-pytest-timeout: https://cgit.freebsd.org/ports/tree/devel/py-pytest-timeout/Makefile
- devel/pyside2: https://cgit.freebsd.org/ports/tree/devel/pyside2/Makefile
- devel/shiboken2: https://cgit.freebsd.org/ports/tree/devel/shiboken2/Makefile
- devel/py-setuptools: https://cgit.freebsd.org/ports/tree/devel/py-setuptools/Makefile
- devel/py-typing-extensions: https://cgit.freebsd.org/ports/tree/devel/py-typing-extensions/Makefile
- devel/py-pyyaml: https://cgit.freebsd.org/ports/tree/devel/py-pyyaml/Makefile

### CODE REVIEW

**Comment by cottsay** [comment]
To verify these rules I implemented `rosdep_repo_check` support for FreeBSD.

I'm not ready to submit that change yet for a couple of reasons:
1. PyYAML takes over 2 minutes to parse the metadata, and that's just too long for these checks
2. I needed to change PyYAML to be able to "stream" the package list so that the check ends when all of the packages are found. While I did get this working, I'm not very fond of how deep into PyYAML I had to reach to make it happen.

File: rosdep/base.yaml [hunk 1]
```diff
@@ -2596,6 +2603,7 @@ libatomic:
   arch: [gcc-libs]
   debian: [libatomic1]
   fedora: [libatomic]
+  freebsd: [gcc]
```
Comment (sloretz) [inline rosdep/base.yaml:2606]: I couldn't figure out if libatomic was provided by gcc

**Review by sloretz — APPROVED** [review]
yolo

File: rosdep/base.yaml (hunk 1 above)
Comment (cottsay) [reply rosdep/base.yaml:2606]: I think I'll just remove this rule. It seems that SOME of the gcc versions in FreeBSD have libatomic.so, but not the DEFAULT one.

**Pull request merged**